### PR TITLE
allow for traversing folders in host list

### PR DESCRIPTION
**Pull request by warroyo**
also split out host listing into private function.

### CODE REVIEW

File: lib/chef/knife/vsphere_hosts_list.rb [hunk 1]
```diff
@@ -14,8 +14,10 @@ class Chef::Knife::VsphereHostsList < Chef::Knife::BaseVsphereCommand
          description: 'Target pool'
 
   def find_pools(folder, poolname = nil)
-    pools = folder.children.find_all.select { |p| p.is_a?(RbVmomi::VIM::ComputeResource) || p.is_a?(RbVmomi::VIM::ResourcePool) }
-    poolname.nil? ? pools : pools.select { |p| p.name == poolname }
+    pools = traverse_folders_for_pools(folder)
+    clusters = traverse_folders_for_computeresources(folder)
+    cluster_pool = clusters + pools
+    poolname.nil? ? cluster_pool : cluster_pool.select { |p| p.name == poolname }
   end
```
Comment (swalberg) [inline lib/chef/knife/vsphere_hosts_list.rb]: Is this method something that could be extracted? I thought I remember it being in your last PR, too.

Comment (warroyo) [inline lib/chef/knife/vsphere_hosts_list.rb]: thats a good point, it is doing the exact same thing. I will move this to the base class.

**Comment by swalberg** [comment]
Awesome, thanks!

**Comment by swalberg** [comment]
Just released 1.2.22, so your improvements are now available to the world.

**Comment by warroyo** [comment]
Awesome, Thank you!

**Pull request merged**